### PR TITLE
Avoid the FocusTrap wrapping div by default

### LINKED ISSUE #23: Focused wrapper div gets border
&lt;HotKeys/> wraps its children into a &lt;div tabIndex="-1"/>. When the user clicks the form outside of any control, the whole form gets a weird focused border created by the browser (Chrome has a particularly visible one).

Is there a way to prevent this?

Some probably stupid ideas:
- Push the border somehow on the very limits of the viewport to make it less eye-catching.
- Add code to the &lt;div/> to drop focus after it receives it.

### PR DESCRIPTION
**Pull request by opichals**
The onFocus/onBlur handling is delegated to the only HotKeys child element which gets elementClone()d in the render method.

Fixes #23

### CODE REVIEW

**Comment by chrisui** [comment]
`FocusTrap` should stick around. It's an important abstraction, it manages focus state, hotkeys manage key maps and handlers and hook into focus state.

I agree there should be no need to render an extra wrapping `div` though but this can be achieved another way which I'd be happy to accept.

`FocusTrap` should only accept one child so it can simply clone it's children as you have done here with `HotKeys`. 

`HotKeys` can then have an api like `<HotKeys component="div" />` if you want so you can control what it renders as (note that component *must* forward `onFocus` and `onBlur` props).

Note the HoC api might (subjectively) achieve this much more elegantly (again would still need to forward the mentioned props).

**Comment by opichals** [comment]
Thanks for the feedback. I understand, I have changed the PR so that the FocusTrap is still there.

The API change is that it FocusTrap defaults to not wrapping its contents and therefore requiring a single element children.

**Comment by MLoughry** [comment]
I'm looking at integrating this package into my project, but this issue is a major pain point. Is there any reason not to merge this PR and generate a new release?

File: lib/FocusTrap.js [hunk 1]
```diff
@@ -11,22 +13,52 @@ const FocusTrap = React.createClass({
 
   getDefaultProps() {
     return {
-      component: 'div'
+      component: DOCUMENT_FRAGMENT
```
Comment (chrisui) [inline lib/FocusTrap.js]: is `"documentFragment"` a new React thing? Not familiar.

File: lib/FocusTrap.js [hunk 2]
```diff
@@ -11,22 +13,52 @@ const FocusTrap = React.createClass({
...
+      onFocus(...args);
+    }
+  },
+
+  onBlur(...args) {
+    const {children, onBlur} = this.props;
+
+    const child = React.Children.only(children);
+    if (child.onBlur) {
+      child.onBlur(...args);
```
Comment (chrisui) [inline lib/FocusTrap.js]: Calling methods on children is a bit anti-react as it breaks the direction of data.

**Comment by chrisui** [comment]
I'm gonna spend a little time on https://github.com/chrisui/react-hotkeys/issues/24 tomorrow at work (finally) and I'll incorporate the work here.

Can sync post that investigation but hopefully I'll have something finished and released this weekend.

Serious apologies for the lack of activity around here!

**Comment by MikeTaylor** [comment]
Can I raise a hand and ask for this? We really need `<HotKeys>` to not wrap its children, as that breaks assumptions about nesting that our CSS makes.

**Comment by MikeTaylor** [comment]
Meanwhile, I am trying to get by using
```
<HotKeys component="documentFragment" keyMap={globalKeyMap} handlers={handlers}>
```
but that doesn't avoid wrapping the children, as the code in this PR had led me to hope; it just means they get wrapped in a `<documentfragment tabindex="-1">` instead of `<div tabindex="-1">`.

Is that expected?

**Comment by greena13** [comment]
It's looking very likely that we will be merging in #61 in favour of this pull request. Thank you for your work on this, as it no doubt informed #61 and helped arrive at an alternative solution.

**Comment by opichals** [comment]
It is good that somebody took the responsibility for the lib maintenance.

It is also good to note that there should be no need for wrapping anymore with [React v16](https://reactjs.org/blog/2017/09/26/react-v16.0.html).

**Pull request closed without merging**